lib/model.js

```diff
@@ -5001,6 +5002,15 @@ Model.__subclass = function subclass(conn, schema, collection) {
 
 Model.recompileSchema = function recompileSchema() {
   this.prototype.$__setSchema(this.schema);
+
+  if (this.schema._applyDiscriminators != null) {
+    for (const disc of this.schema._applyDiscriminators.keys()) {
+      this.discriminator(disc, this.schema._applyDiscriminators.get(disc));
+    }
+  }
+
+  const overwriteExisting = true;
+  applyEmbeddedDiscriminators(this.schema, new WeakSet(), overwriteExisting);
```

Comment:
and specific reason on why `overwriteExisting` is not inlined?

Reply:
I was thinking about doing this for readability, but given that there's only 1 boolean param I don't think it helps readability much. I removed that in [1e8753f](https://github.com/Automattic/mongoose/pull/14500/commits/1e8753fa6a94fc8649ad5cb27b5247d4d4f3c459)